src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs

```diff
@@ -2,344 +2,98 @@
-
     public string? VSTestSessionCorrelationId { get; set; }
 
-    public override bool Execute()
+    protected override string? ToolName
     {
-        var traceEnabledValue = Environment.GetEnvironmentVariable("VSTEST_BUILD_TRACE");
-        Tracing.traceEnabled = !traceEnabledValue.IsNullOrEmpty() && traceEnabledValue.Equals("1", StringComparison.OrdinalIgnoreCase);
-
-        var debugEnabled = Environment.GetEnvironmentVariable("VSTEST_BUILD_DEBUG");
-        if (!debugEnabled.IsNullOrEmpty() && debugEnabled.Equals("1", StringComparison.Ordinal))
+        get
         {
-            Console.WriteLine("Waiting for debugger attach...");
-
-            var currentProcess = Process.GetCurrentProcess();
-            Console.WriteLine(string.Format("Process Id: {0}, Name: {1}", currentProcess.Id, currentProcess.ProcessName));
-
-            while (!Debugger.IsAttached)
-            {
-                Thread.Sleep(1000);
-            }
-
-            Debugger.Break();
-        }
-
-        // Avoid logging "Task returned false but did not log an error." on test failure, because we don't
-        // write MSBuild error. https://github.com/dotnet/msbuild/blob/51a1071f8871e0c93afbaf1b2ac2c9e59c7b6491/src/Framework/IBuildEngine7.cs#L12
-        var allowfailureWithoutError = BuildEngine.GetType().GetProperty("AllowFailureWithoutError");
-        allowfailureWithoutError?.SetValue(BuildEngine, true);
-
-        _vsTestForwardingApp = new VSTestForwardingApp(VSTestConsolePath, CreateArgument());
-        if (!VSTestFramework.IsNullOrEmpty())
-        {
-            Console.WriteLine(Resources.Resources.TestRunningSummary, TestFileFullPath, VSTestFramework);
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                return "dotnet.exe";
+            else
+                return "dotnet";
         }
-
-        return _vsTestForwardingApp.Execute() == 0;
     }
 
-    public void Cancel()
+    public VSTestTask()
     {
-        Tracing.Trace("VSTest: Killing the process...");
-        _vsTestForwardingApp?.Cancel();
+        LogStandardErrorAsError = true;
+        StandardOutputImportance = "Normal";
     }
 
-    internal IEnumerable<string> CreateArgument()
+    protected override string? GenerateCommandLineCommands()
     {
-        var allArgs = AddArgs();
-
-        // VSTestCLIRunSettings should be last argument in allArgs as vstest.console ignore options after "--"(CLIRunSettings option).
-        AddCliRunSettingsArgs(allArgs);
-
-        return allArgs;
-    }
-
-    private void AddCliRunSettingsArgs(List<string> allArgs)
-    {
-        if (VSTestCLIRunSettings != null && VSTestCLIRunSettings.Length > 0)
-        {
-            allArgs.Add("--");
-            foreach (var arg in VSTestCLIRunSettings)
-            {
-                allArgs.Add(ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(arg));
-            }
-        }
+        return this.CreateCommandLineArguments();
```

Comment:
Nit: 
```suggestion
        return CreateCommandLineArguments();
```

Reply:
This is an extension method, in order to keep as much functionality shared between the tasks as possible. So I think `this` is necessary here.
It might be reworked as a simple static method instead?

Reply:
I think what @Evangelink means is this:

```suggestion
        return CreateCommandLineArguments(this);
```

But 

> It might be reworked as a simple static method instead?

I might prefer this instead!

Reply:
Sorry I missed it was an extension method. Yeah seems like it would be better as static method.